File: Ejercicio 016/empresa.java
class Empleado {
    String nombre,id;

    public Empleado(String n,String i){
        nombre=n;
        id=i;
    }

    public void rol(){}
}

class Desarrollador extends Empleado{
    public Desarrollador(String n,String i){ super(n,i); }
    public void rol(){ System.out.println("Desarrolla software"); }
}

class Tester extends Empleado{
    public Tester(String n,String i){ super(n,i); }
    public void rol(){ System.out.println("Prueba aplicaciones"); }
}

class ProjectManager extends Empleado{
    public ProjectManager(String n,String i){ super(n,i); }
    public void rol(){ System.out.println("Gestiona proyectos"); }
}

public class Main {
    public static void main(String[] args){

        Empleado[] lista={
            new Desarrollador("Ana","01"),
            new Tester("Luis","02"),
            new ProjectManager("Carlos","03")
        };

        for(Empleado e:lista){
            System.out.println(e.getClass().getSimpleName());
            e.rol();
            System.out.println("-----");
        }
    }
}

File: Ejercicio 016/empresa.py
class Empleado:
    def __init__(self,nombre,id):
        self.nombre=nombre
        self.id=id

    def rol(self):
        pass


class Desarrollador(Empleado):
    def rol(self):
        print("Desarrolla software")


class Tester(Empleado):
    def rol(self):
        print("Prueba aplicaciones")


class ProjectManager(Empleado):
    def rol(self):
        print("Gestiona proyectos")



lista=[
    Desarrollador("Ana","01"),
    Tester("Luis","02"),
    ProjectManager("Carlos","03")
]

for e in lista:
    print(type(e).__name__)
    e.rol()
    print("-----")
